code/game/turfs/simulated/floor.dm

```diff
@@ -666,26 +665,20 @@ turf/simulated/floor/update_icon()
 						update_icon()
 
 /turf/simulated/proc/wet(delay = 800, slipperiness = TURF_WET_WATER)
-	if(wet >= slipperiness)
-		return
-	wet = slipperiness
-	if(wet_overlay)
-		overlays -= wet_overlay
-		wet_overlay = null
-	wet_overlay = w_overlays["wet"]
-	overlays += wet_overlay
-	spawn(delay)
-		if(!istype(src, /turf/simulated)) //Because turfs don't get deleted, they change, adapt, transform, evolve and deform. they are one and they are all.
-			return
-		dry(slipperiness)
+	var/obj/effect/overlay/puddle/P = locate(/obj/effect/overlay/puddle) in src
+	if(P)
+		if(slipperiness > P.wet)
+			P.wet = slipperiness
+			P.lifespan += delay
```

Comment:
Super minor but it seems like you can wet a floor then lube it for double the lube duration, it might even do that on its own if you use water and lube in the same spray

Reply:
`max(lifespan, delay)` maybe?